x-pack/plugins/viz_editor/public/common/components/operation_editor/operation_definitions.tsx

```diff
@@ -0,0 +1,431 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiFieldNumber, EuiForm, EuiFormRow, EuiIcon, EuiSelect } from '@elastic/eui';
+import React from 'react';
+import {
+  AvgOperation,
+  CardinalityOperation,
+  ColumnOperation,
+  CountOperation,
+  DatasourceField,
+  DateHistogramOperation,
+  FieldOperation,
+  SelectOperation,
+  SumOperation,
+  TermsOperation,
+  WindowFunction,
+  WindowOperation,
+} from '../../../../common';
+
+export interface OperationEditorProps {
+  children: any;
+  column: SelectOperation;
+  onColumnChange: (newColumn: SelectOperation) => void;
+  visModel: any;
+  allowedOperations?: Array<SelectOperation['operation']>;
```

Comment:
1. I've left a few comments about passing around operation names like this, because I don't think it's maintainable. What are your thoughts on helping this be maintainable?

2. If you are going to pass around operation names, please use `SelectOperator[]`

Reply:
`SelectOperation['operation']` refers to the same collections of string literals in the end, but I agree with you, we should only have one of them - my bad. I would actually like something like `type SelectOperator = SelectOperation['operation']`, because it means you don't have to specify each operator twice (in the SelectOperator type and in the respective Operation definition,`export type SelectOperator ='column' | ...` and `interface ColumnOperation extends FieldOperation {
  operation: 'column';`)

Totally open for discussions about this though.

Regarding maintainability - see comment below.